lib/matplotlib/axes/_axes.py

```diff
@@ -3061,18 +3064,17 @@ def get_next_color():
             w = mpatches.Wedge((x, y), radius, 360. * min(theta1, theta2),
                                360. * max(theta1, theta2),
                                facecolor=get_next_color(),
-                               **wedgeprops)
+                               clip_on=False,
+                               label=label)
+            w.set(**wedgeprops)
             slices.append(w)
             self.add_patch(w)
-            w.set_label(label)
 
             if shadow:
-                # make sure to add a shadow after the call to
-                # add_patch so the figure and transform props will be
-                # set
+                # Make sure to add a shadow after the call to add_patch so the
+                # figure and transform props will be set.
                 shad = mpatches.Shadow(w, -0.02, -0.02)
-                shad.set_zorder(0.9 * w.get_zorder())
-                shad.set_label('_nolegend_')
+                shad.set(zorder=0.9 * w.get_zorder(), label='_nolegend_')
```

Comment:
Any reason these parameters are not passed directly to `Shadow`?

I'm also wondering if we need to specify the label, because by default an Artist should not have any label.

Reply:
This doesn't work, likely because of bad interaction with https://github.com/matplotlib/matplotlib/pull/16098.
Also by default Shadow inherits the parent patch (the Wedge)'s props, including the label.  Again perhaps this should be changed on Shadow, but heh.

Reply:
Ok, was here before. I'll close my eyes and move on.